hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/config/StreamFileHealthIndicator.java

```diff
@@ -0,0 +1,163 @@
+package com.hedera.mirror.importer.config;
+
+/*-
+ * ‌
+ * Hedera Mirror Node
+ * ​
+ * Copyright (C) 2019 - 2021 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+import static com.hedera.mirror.importer.downloader.Downloader.STREAM_CLOSE_LATENCY_METRIC_NAME;
+import static com.hedera.mirror.importer.parser.StreamFileParser.STREAM_PARSE_DURATION_METRIC_NAME;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tags;
+import io.micrometer.core.instrument.Timer;
+import java.time.Instant;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.springframework.boot.actuate.health.Health;
+import org.springframework.boot.actuate.health.HealthIndicator;
+import org.springframework.boot.actuate.health.Status;
+
+import com.hedera.mirror.importer.MirrorProperties;
+import com.hedera.mirror.importer.parser.ParserProperties;
+
+@Log4j2
+@RequiredArgsConstructor
+public class StreamFileHealthIndicator implements HealthIndicator {
+    private static final String COUNT_KEY = "count";
+    private static final String LAST_CHECK_KEY = "lastCheck";
+    private static final String REASON_KEY = "reason";
+
+    private final ParserProperties parserProperty;
+    private final MeterRegistry meterRegistry;
+    private final MirrorProperties mirrorProperties;
+    private final Health parsingDisabledHealth = Health
+            .unknown()
+            .withDetail(REASON_KEY, "Parsing is disabled")
+            .build();
+    private final Health missingParseDurationTimerHealth = Health
+            .unknown()
+            .withDetail(REASON_KEY, String.format("%s timer is missing", STREAM_PARSE_DURATION_METRIC_NAME))
+            .build();
+    private final Health missingStreamCloseTimerHealth = Health
+            .unknown()
+            .withDetail(REASON_KEY, String.format("%s timer is missing", STREAM_CLOSE_LATENCY_METRIC_NAME))
+            .build();
```

Comment:
Should be static and can shorten with a helper method. Can also collapse the the two missing timers that way:
```suggestion
    private static final Health HEALTH_DISABLED = unknownHealth("Parsing is disabled");
    private static final Health HEALTH_MISSING_PARSE_TIMER = unknownHealth(STREAM_PARSE_DURATION_METRIC_NAME + " timer is missing");
    private static final Health HEALTH_MISSING_CLOSE_TIMER = unknownHealth(STREAM_CLOSE_LATENCY_METRIC_NAME + " timer is missing");

    private static Health unknownHealth(String reason) {
    ...
    }
```